.github/workflows/release.yml

```diff
@@ -0,0 +1,143 @@
+# This workflow will build a Java project with Maven, and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-java-with-maven
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: Release aissemble
+
+on:
+  workflow_dispatch:
+    inputs:
+      releaseVersion:
+        description: "The version you want to release aiSSEMBLE on."
+        required: true
+        type: string
+      developmentVersion:
+        description: "The version that dev should be on after this release."
+        required: true
+        type: string
+      releaseBranch:
+        description: "OPTIONAL: The branch to use for running the release."
+        required: false
+        type: string
+      existingVersion:
+        description: "OPTIONAL: The existing aiSSEMBLE version. Defaults to patchVersion-SNAPSHOT"
+        required: false
+        type: string
+      dryRun:
+        description: "OPTIONAL: Used to run the release job as a dry run, skipping the deploy phase"
+        required: false
+        default: false
+        type: boolean
+  push:
+    branches: [ "3-add-release" ]
+
+jobs:
+  build:
+
+    runs-on: arc-runner-set-aissemble
+    env:
+      DOCKER_CONFIG: /home/runner/.docker
+      RELEASE_VERSION: ${{ inputs.releaseVersion || '1.9.0-rc2' }}
+      NEXT_DEVELOPMENT_VERSION: ${{ inputs.developmentVersion || '1.10.0-SNAPSHOT' }}
+      RELEASE_BRANCH: ${{ inputs.releaseBranch }}
+      EXISTING_VERSION: ${{ inputs.existingVersion }}
+      DRY_RUN: ${{ inputs.dryRun || true }}
+
+    steps:
+      - name: Import GPG key
+        uses: crazy-max/ghaction-import-gpg@v6
+        with:
+          gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
+          passphrase: ${{ secrets.GPG_PRIVATE_KEY_PASSPHRASE }}
+      - name: Install required packages
+        run: |
+          sudo apt-get update
+          sudo apt install -y build-essential libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev curl \
+            git libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev zip unzip \
+            libpython3.11
+      - name: Install Python
+        uses: gabrielfalcao/pyenv-action@v18
+        with:
+          default: 3.11.4
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+      - name: Install Helm
+        run: |
+          curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3
+          chmod 700 get_helm.sh
+          ./get_helm.sh
+      - name: Install Helm Unittest Plugin
```

Comment:
S: This is just a copy of what we're doing in `build.yaml` right? Strongly suggest we pull these out into a common place if so.  We're already introducing the complexity of composite actions, and this is the core use case for them.

Reply:
Added instal dependencies composite action.